### PR TITLE
Vebt-839 and Vebt-840: 22-10216 step 1 and 2

### PR DESCRIPTION
**Pull request by fatmakhan0395**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
- Added 22-10216 - STEP 1: Institutions details
- Added unit and cypress tests
- Added 22-10216 - STEP 2: Student Ratio Calculation
- Added unit and cypress tests

- _(Summarize the changes that have been made to the platform)_
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
- _(Which team do you work for, does your team own the maintenance of this component?)_
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)
vebt-839:
As a...
 
VEBT Developer 
 
I want...
 
To build STEP 1 of the 22-10216
 
So that...
 
Form development can be completed
 
Acceptance Criteria
About page and Intro page will be confirmed later, at this time please check with Dev lead about navigation and Form start page ( follow intro page blank template with link to start form). Form name - H1 - Request exemption from the 85/15 Rule reporting requirements
 
This will be STEP 1 alone. FE fields will be created, but will need to have new tickets to manage BE linkup ( to GIDS) for fields such as Facility Code. Please review mockups attached. 
 - STEP 1 of 4: Institution Details
 -- Institution Name ({color:#de350b}*REQ{color})
 -- Facility Code ({color:#de350b}*REQ{color}) ({color:#ff8b00}NOTE:- This is require auto confirmation if its a accredited school, if not accredited it will show the next page{color}).
 -- Term Start Date ({color:#de350b}*REQ{color})
 - STEP 1 of 4: Institution Details (Part 2 - this is just for reference)
 -- Yellow pop up box if a school entered is not {color:#ff8b00}accredited. Text should read as "Your school facility code indicates the school is not accredited. In addition to completing VA Form 22-10216, you'll also need to complete and submit VA Form 22-10215. You will be directed to that form after completing this one."{color}

Vebt-840:
As a...

VEBT Developer 

I want…

To build STEP 2 of the 22-10216

So that…

Form development can be completed

Acceptance Criteria

STEP 1 of 4: Institution Details
Number of VA beneficiary students (*required)
Total number of students (*required)
VA beneficiary students percentage 

(calculated)

- - - (Note:- This is left blank until the top two fields are entered, this is auto calculated).
How is this calculated? ( this is a drop down, collapsed by default). 

"Number of VA beneficiary students divided by Total number of students) multiplied by 100 = Percentage of VA beneficiary students
If this percentage seems incorrect, please check the numbers you entered above."

Date of calculation (*required) ( Date field) 
Other details :- 

Technical Details: Note for Dev's :- Unit tests are included as part of this ticket. **

- _Link to ticket created in va.gov-team repo_
- https://jira.devops.va.gov/browse/VEBT-839
- https://jira.devops.va.gov/browse/VEBT-840
department-of-veterans-affairs/va.gov-team#0000
- _Link to previous change of the code/bug (if applicable)_
department-of-veterans-affairs/vets-website#0000
- _Link to epic if not included in ticket_
department-of-veterans-affairs/va.gov-team#0000

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

image

image

image

image

image

image

image

Mobile:
image

image

image

image

image

image

image

image

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/edu-benefits/10216/components/PercentageCalc.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { calculatedPercentage } from '../utilities';
+
+const gaRatioCalculationText = () => {
+  window.dataLayer.push({
+    event: 'edu-0994--form-help-text-clicked',
```
Comment (GovCIOLiz) [inline src/applications/edu-benefits/10216/components/PercentageCalc.jsx:8]: wrong form analytics prefix

File: src/applications/edu-benefits/10216/components/PercentageCalc.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    <div className="schemaform-field-template">
+      <span className="vads-u-font-weight--bold vads-u-font-size--lg">
+        {calculatedPercentage(formData)}
+      </span>
+      <va-additional-info
+        class="vads-u-margin-top--1p5"
+        trigger="How is this calculated?"
+        onClick={gaRatioCalculationText}
+      >
+        <span>
```
Comment (GovCIOLiz) [inline src/applications/edu-benefits/10216/components/PercentageCalc.jsx:23]: wrapping in span vs fragment might add some padding-left here we don't want, please check

File: src/applications/edu-benefits/10216/pages/studentRatioCalc.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  },
+  schema: {
+    type: 'object',
+    properties: {
+      studentRatioCalcChapter: {
+        type: 'object',
+        properties: {
+          beneficiaryStudent: numberSchema,
+          numOfStudent: numberSchema,
+          studentPercentageCalc: {
```
Comment (GovCIOLiz) [inline src/applications/edu-benefits/10216/pages/studentRatioCalc.js:57]: If you do `'view:studentPercentageCalc': { type: 'object', properties: {} }` it wont end up in the data and you can remove some code in the submit-transformer. And the corresponding uiSchema would be `view:studentPercentageCalc` with whats already included in that object

**Review by GovCIOLiz — DISMISSED** [review]

**Comment by wafimohamed** [comment]
not needed. we created an other PR

**Pull request closed without merging**